doctl k8s cluster get <clustername> not working for cluster with long name

when I am creating a cluster with a long name "doctl k8s cluster get $CLUSTER" not working, but in list command it is showing the cluster

doctl k8s cluster get $CLUSTER

Error: GET https://api.digitalocean.com/v2/kubernetes/clusters/f8291060b73f4fa7b60586fe51a1d862: 404 cluster not found


doctl k8s cluster list

ID                                      Name                                Region    Version        Status     Node Pools
c4fbbc63-9ebf-4b0e-8ec2-4df84549e3d4    f8291060b73f4fa7b60586fe51a1d862    blr1      1.14.1-do.3    running    f8291060b73f4fa7b60586fe51a1d862-default-pool

Hi @praveenvp07 Thanks for the report.

We allow up to 64 characters in the name for a Kubernetes cluster,  and `doctl k8s cluster get` does work with a name that long:

```
$ doctl k8s cluster get thisisakubernetesclusterwithaverrylongnameaslongasweallowittobe
ID                                      Name                                                               Region    Version        Auto Upgrade    Status          Endpoint    IPv4    Cluster Subnet    Service Subnet    Tags                                            Created At                       Updated At                       Node Pools
21464f77-42fc-4a32-9aa4-a101843e94c0    thisisakubernetesclusterwithaverrylongnameaslongasweallowittobe    nyc1      1.14.1-do.3    false           provisioning                        10.244.0.0/16     10.245.0.0/16     k8s,k8s:21464f77-42fc-4a32-9aa4-a101843e94c0    2019-06-06 19:28:09 +0000 UTC    2019-06-06 19:28:19 +0000 UTC    pool-v5ju1cmsh
```

The issue here seem to be that we use `uuid.Parse` to determine if the user supplied a name or an ID as the argument:

https://github.com/digitalocean/doctl/blob/master/commands/kubernetes.go#L1515-L1517

This is incorrectly determining that your cluster's name is actually its ID. Here's a test case demonstrating the issue:

```
func TestClusterByIDorName(t *testing.T) {
	tests := []struct {
		name string
		want bool
	}{
		{
			name: "21464f77-42fc-4a32-9aa4-a101843e94c0",
			want: true,
		},
		{
			name: "my-cluster",
			want: false,
		},
		{
			name: "f8291060b73f4fa7b60586fe51a1d862",
			want: false,
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			assert.EqualValues(t, tt.want, looksLikeUUID(tt.name))
		})
	}
}
```

Result:

```
--- FAIL: TestClusterByIDorName (0.00s)
    --- FAIL: TestClusterByIDorName/f8291060b73f4fa7b60586fe51a1d862 (0.00s)
        kubernetes_test.go:876: 
            	Error Trace:	kubernetes_test.go:876
            	Error:      	Not equal: 
            	            	expected: false
            	            	actual  : true
            	Test:       	TestClusterByIDorName/f8291060b73f4fa7b60586fe51a1d862
FAIL
```

--- a/commands/kubernetes.go
+++ b/commands/kubernetes.go
@@ -1679,7 +1679,7 @@ func nodeByName(name string, nodes []*godo.KubernetesNode) (*godo.KubernetesNode
 }
 
 func looksLikeUUID(str string) bool {
-	r := regexp.MustCompile("^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[14][a-fA-F0-9]{3}-[8|9|aA|bB][a-fA-F0-9]{3}-[a-fA-F0-9]{12}$")
+	r := regexp.MustCompile("^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[1-4][a-fA-F0-9]{3}-[8|9|aA|bB][a-fA-F0-9]{3}-[a-fA-F0-9]{12}$")
 	return r.MatchString(str)
 }
 

--- a/commands/kubernetes.go
+++ b/commands/kubernetes.go
@@ -20,6 +20,7 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+	"regexp"
 	"sort"
 	"strconv"
 	"strings"
@@ -30,7 +31,6 @@ import (
 	"github.com/digitalocean/doctl/commands/displayers"
 	"github.com/digitalocean/doctl/do"
 	"github.com/digitalocean/godo"
-	"github.com/google/uuid"
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
 
@@ -1679,8 +1679,8 @@ func nodeByName(name string, nodes []*godo.KubernetesNode) (*godo.KubernetesNode
 }
 
 func looksLikeUUID(str string) bool {
-	_, err := uuid.Parse(str)
-	return err == nil
+	r := regexp.MustCompile("^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-4[a-fA-F0-9]{3}-[8|9|aA|bB][a-fA-F0-9]{3}-[a-fA-F0-9]{12}$")
+	return r.MatchString(str)
 }
 
 func getVersionOrLatest(c *CmdConfig) (string, error) {

--- a/commands/kubernetes.go
+++ b/commands/kubernetes.go
@@ -1679,7 +1679,7 @@ func nodeByName(name string, nodes []*godo.KubernetesNode) (*godo.KubernetesNode
 }
 
 func looksLikeUUID(str string) bool {
-	r := regexp.MustCompile("^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-4[a-fA-F0-9]{3}-[8|9|aA|bB][a-fA-F0-9]{3}-[a-fA-F0-9]{12}$")
+	r := regexp.MustCompile("^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[14][a-fA-F0-9]{3}-[8|9|aA|bB][a-fA-F0-9]{3}-[a-fA-F0-9]{12}$")
 	return r.MatchString(str)
 }
 

--- a/commands/kubernetes_test.go
+++ b/commands/kubernetes_test.go
@@ -1001,24 +1001,33 @@ func TestLatestVersionForUpgrade(t *testing.T) {
 func Test_looksLikeUUID(t *testing.T) {
 	tests := []struct {
 		name string
+		arg  string
 		want bool
 	}{
 		{
-			name: "21464f77-42fc-4a32-9aa4-a101843e94c0",
+			name: "UUIDv1",
+			arg:  "a235f190-01a4-11ea-af17-4989e8155574",
+			want: true,
+		},
+		{
+			name: "UUIDv4",
+			arg:  "21464f77-42fc-4a32-9aa4-a101843e94c0",
 			want: true,
 		},
 		{
 			name: "my-cluster",
+			arg:  "my-cluster",
 			want: false,
 		},
 		{
 			name: "f8291060b73f4fa7b60586fe51a1d862",
+			arg:  "f8291060b73f4fa7b60586fe51a1d862",
 			want: false,
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			if got := looksLikeUUID(tt.name); got != tt.want {
+			if got := looksLikeUUID(tt.arg); got != tt.want {
 				t.Errorf("looksLikeUUID() = %v, want %v", got, tt.want)
 			}
 		})

--- a/commands/kubernetes_test.go
+++ b/commands/kubernetes_test.go
@@ -9,7 +9,6 @@ import (
 	"github.com/digitalocean/godo"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
-
 	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 )
 
@@ -994,6 +993,33 @@ func TestLatestVersionForUpgrade(t *testing.T) {
 			} else {
 				require.True(t, found)
 				require.Equal(t, tt.want, slug)
+			}
+		})
+	}
+}
+
+func Test_looksLikeUUID(t *testing.T) {
+	tests := []struct {
+		name string
+		want bool
+	}{
+		{
+			name: "21464f77-42fc-4a32-9aa4-a101843e94c0",
+			want: true,
+		},
+		{
+			name: "my-cluster",
+			want: false,
+		},
+		{
+			name: "f8291060b73f4fa7b60586fe51a1d862",
+			want: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := looksLikeUUID(tt.name); got != tt.want {
+				t.Errorf("looksLikeUUID() = %v, want %v", got, tt.want)
 			}
 		})
 	}

